python/xir/program.py

```diff
@@ -211,165 +213,245 @@ class XIRProgram:
+        """Returns the declarations in the XIR program.
 
         Returns:
-            Dict: declared scipt-level options
+            Mapping[str, List[Declaration]]: dictionary of declarations sorted
+                into the following keys: 'gate', 'func', 'output' and 'operator'.
         """
-        if self.use_floats:
-            return get_floats(self._options)
-        return self._options
+        return self._declarations
 
     @property
-    def statements(self) -> List[Statement]:
-        """Statements in the program
+    def gates(self) -> Mapping[str, Mapping[str, Sequence]]:
+        """Returns the gates in the XIR program.
 
         Returns:
-            list[Statement]: a list of all statements
+            Mapping[str, Mapping[str, Sequence]]: dictionary of gates, each gate
+                consisting of a name and a dictionary with the following keys:
+                'parameters', 'wires' and 'statements'
         """
-        return self._statements
+        return self._gates
 
     @property
-    def declarations(self) -> Dict[str, List]:
-        """Declarations in the program
+    def includes(self) -> Iterator[str]:
+        """Returns the included XIR modules used by the XIR program.
 
         Returns:
-            dict[str, list]: a dictionary of all declarations sorted into the following keys:
-            'gate', 'func', 'output' and 'operator'
+            Iterator[str]: iterator over the included XIR modules
         """
-        return self._declarations
+        return iter(self._includes)
 
     @property
-    def gates(self) -> Dict[str, Dict[str, Sequence]]:
-        """All user-defined gates in the program
+    def operators(self) -> Mapping[str, Mapping[str, Sequence]]:
+        """Returns the operators in the XIR program.
 
         Returns:
-            dict[str, dict]: a dictionary of all user-defined gates, each gate consisting of a name
-            as well as a dictionary containing parameters, wires and statements
+            Mapping[str, Mapping[str, Sequence]]: dictionary of operators, each
+                operator consisting of a name and a dictionary with the following
+                keys: 'parameters', 'wires' and 'statements'
         """
-        return self._gates
+        return self._operators
 
     @property
-    def operators(self) -> Dict[str, Dict[str, Sequence]]:
-        """All user-defined operators in the program
+    def options(self) -> Mapping[str, Any]:
+        """Returns the script-level options declared in the XIR program.
 
         Returns:
-            dict[str, dict]: a dictionary of all user-defined operators, each operator consisting of
-            a name as well as a dictionary containing parameters, wires and statements
+            Mapping[str, Any]: declared scipt-level options
         """
-        return self._operators
+        return get_floats(self._options) if self.use_floats else self._options
```

Comment:
I don't mind this change per se, but I prefer using several lines (as before). It being slightly cleaner and clearer to understand with a glance.